clients/segments/segments.go

```diff
@@ -113,27 +113,16 @@ func (c Client) GetAll(ctx context.Context) ([]Response, error) {
 	return result, nil
 }
 
-// Upsert creates a new entry of segment on server in case a configuration with an ID doesn't already exist on server. If exists, it updates it.
-// If the method is provided with id = "" creat will be called.
-func (c Client) Upsert(ctx context.Context, id string, data []byte) (Response, error) {
-	//To allow passing empty id for direct call to create without GET
-	existing := &http.Response{StatusCode: http.StatusNotFound}
-	var err error
-	if id != "" {
-		existing, err = c.client.Get(ctx, id, rest.RequestOptions{CustomShouldRetryFunc: rest.RetryIfTooManyRequests})
-		closeBody(existing)
-		if err != nil {
-			return Response{}, fmt.Errorf(errMsgWithId, "upsert", id, err)
-		}
+// Update the matching resource on remote, before the update get is called to ge the version and owner field to be added
+func (c Client) Update(ctx context.Context, id string, data []byte) (Response, error) {
+	if id == "" {
```

Comment:
The rest of the code is not checking it. It relies on that the core client will do that (e.g., openpipeline, buckets)
Maybe we should discuss it in TecTime.

Reply:
No need for TechTime, i just didn't see the get checks for empty id. :)